crates/aptos-node-identity/src/lib.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{format_err, Result};
+use aptos_types::{chain_id::ChainId, PeerId};
+use once_cell::sync::OnceCell;
+use std::sync::Arc;
+
+/// The global [AptosNodeIdentity]
+static APTOS_NODE_IDENTITY: OnceCell<Arc<AptosNodeIdentity>> = OnceCell::new();
+
+/// Structure that holds information related to a node's identity
+pub struct AptosNodeIdentity {
+    pub chain_id: OnceCell<ChainId>,
+    pub peer_id: Option<PeerId>,
+    pub peer_id_str: Option<String>,
+    pub hostname: Option<String>,
+}
+
+/// Initializes the [AptosNodeIdentity] using the provided [NodeConfig] and
+/// sets it globally exactly once.
+pub fn init(peer_id: Option<PeerId>) -> Result<()> {
+    let identity = AptosNodeIdentity {
+        chain_id: OnceCell::new(),
+        peer_id,
+        peer_id_str: peer_id.map(|id| id.to_string()),
+        hostname: hostname::get()
+            .ok()
+            .and_then(|name| name.into_string().ok()),
+    };
+
+    APTOS_NODE_IDENTITY
+        .set(Arc::new(identity))
+        .map_err(|_| format_err!("APTOS_NODE_IDENTITY was already set"))
+}
+
+pub fn set_chain_id(chain_id: ChainId) -> Result<()> {
+    match APTOS_NODE_IDENTITY.get() {
+        Some(identity) => identity
+            .chain_id
+            .set(chain_id)
+            .map_err(|_| format_err!("chain_id was already set.")),
+        None => Err(format_err!("APTOS_NODE_IDENTITY has not been set yet")),
+    }
+}
+
+/// Returns the [PeerId] from the global [APTOS_NODE_IDENTITY]
+pub fn peer_id() -> Option<PeerId> {
+    APTOS_NODE_IDENTITY
+        .get()
+        .and_then(|identity| identity.peer_id)
+}
+
+/// Returns the [PeerId] as [str] from the global [APTOS_NODE_IDENTITY]
+pub fn peer_id_str() -> Option<&'static str> {
+    APTOS_NODE_IDENTITY
+        .get()
+        .and_then(|identity| identity.peer_id_str.as_deref())
+}
+
+/// Returns the [ChainId] from the global [APTOS_NODE_IDENTITY]
+pub fn chain_id() -> Option<ChainId> {
+    APTOS_NODE_IDENTITY
+        .get()
+        .and_then(|identity| identity.chain_id.get().cloned())
+}
+
+/// Returns the hostname from the global [APTOS_NODE_IDENTITY]
+pub fn hostname() -> Option<String> {
+    APTOS_NODE_IDENTITY
+        .get()
+        .and_then(|identity| identity.hostname.clone())
+}
```

Comment:
nit: can we add a super simple unit test to this crate that tests initialization, setters and getters? Just to make sure it doesn't regress in the future 😄